### PR TITLE
NK-3832 Add method to run exec on docker

### PR DESCRIPTION
**Pull request by arielagenin-zadara**

### CODE REVIEW

File: docker_test_tools/environment.py [hunk 1]
```diff
@@ -388,3 +388,10 @@ def get_container_id(self, name):
     def validate_service_name(self, name):
         if name not in self.services:
             raise ValueError('Invalid service name: %r, must be one of %s' % (name, self.services))
+
+    def run_exec_in_container(self, name, command):
```
Comment (shaitourchin) [inline docker_test_tools/environment.py:392]: Please look at the other functions in this file that work on containers and make this one similar to them.
For example:
1. The other functions start with a docstring
2. The other functions start with a log.debug message

**Review by shaitourchin — CHANGES_REQUESTED** [review]

File: docker_test_tools/environment.py [hunk 2]
```diff
@@ -388,3 +388,21 @@ def get_container_id(self, name):
...
+        """
+        Execute command on container by name
+        :param str name: container name.
+        :param str name: command to run in the container.
```
Comment (shaitourchin) [inline docker_test_tools/environment.py:396]: Please fix the argument name

File: docker_test_tools/environment.py [hunk 3]
```diff
@@ -388,3 +388,21 @@ def get_container_id(self, name):
     def validate_service_name(self, name):
         if name not in self.services:
             raise ValueError('Invalid service name: %r, must be one of %s' % (name, self.services))
+
+    def run_exec_in_container(self, name, command):
+        """
+        Execute command on container by name
```
Comment (shaitourchin) [inline docker_test_tools/environment.py:394]:
```suggestion
        Execute command in container by container name
```

File: docker_test_tools/environment.py [hunk 4]
```diff
@@ -388,3 +388,21 @@ def get_container_id(self, name):
...
+        log.debug("Creating exec instance for container %s", name)
+        container_id = self.get_container_id(name=name)
+        exec_create_output = self.docker_client.exec_create(container_id, command)
+        if not exec_create_output or not exec_create_output.get("Id"):
+            raise RuntimeError("Failed to create exec instance for container %s with command %s" % (name, command))
+        exec_id = exec_create_output.get("Id")
+        log.debug("Starting exec instance for container %s", name)
+        exec_start_output = self.docker_client.exec_start(exec_id)
+        if exec_start_output is None:
+            raise RuntimeError("Failed to start exec instance for container %s with command %s" % (name, command))
```
Comment (shaitourchin) [inline docker_test_tools/environment.py:407]:
```suggestion
            raise RuntimeError("Failed to start exec instance in container %s with command %s" % (name, command))
```

File: docker_test_tools/environment.py [hunk 5]
```diff
@@ -388,3 +388,21 @@ def get_container_id(self, name):
...
+        :param str name: container name.
+        :param str name: command to run in the container.
+        """
+        log.debug("Creating exec instance for container %s", name)
+        container_id = self.get_container_id(name=name)
+        exec_create_output = self.docker_client.exec_create(container_id, command)
+        if not exec_create_output or not exec_create_output.get("Id"):
+            raise RuntimeError("Failed to create exec instance for container %s with command %s" % (name, command))
+        exec_id = exec_create_output.get("Id")
+        log.debug("Starting exec instance for container %s", name)
```
Comment (shaitourchin) [inline docker_test_tools/environment.py:404]:
```suggestion
        log.debug("Starting exec instance in container %s", name)
```

**Review by shaitourchin — CHANGES_REQUESTED** [review]

**Review by shaitourchin — APPROVED** [review]

**Review by grishatsukerman — APPROVED** [review]

**Pull request merged**